config-model/src/main/java/com/yahoo/searchdefinition/derived/RawRankProfile.java

```diff
@@ -50,20 +52,28 @@
     /**
      * Creates a raw rank profile from the given rank profile
      */
+    public RawRankProfile(RankProfile rankProfile, QueryProfileRegistry queryProfiles, ImportedMlModels importedModels, AttributeFields attributeFields, ModelContext.Properties deployProperties) {
+        this.name = rankProfile.getName();
+        compressedProperties = compress(new Deriver(rankProfile, queryProfiles, importedModels, attributeFields, deployProperties).derive());
+    }
+
+    /**
+     * Only for testing
+     */
     public RawRankProfile(RankProfile rankProfile, QueryProfileRegistry queryProfiles, ImportedMlModels importedModels, AttributeFields attributeFields) {
         this.name = rankProfile.getName();
-        compressedProperties = compress(new Deriver(rankProfile, queryProfiles, importedModels, attributeFields).derive());
+        compressedProperties = compress(new Deriver(rankProfile, queryProfiles, importedModels, attributeFields, new TestProperties()).derive());
```

Comment:
See previous comment on `TestProperties`.

Reply:
Also only used for testing.